### PR TITLE
EN/layered arch

### PR DESCRIPTION
**Pull request by ruchit-singh-zs**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,87 +0,0 @@
-linters-settings:
```
Comment (aakanksha-jaiswal-zs) [inline .golangci.yml]: Why have you removed this file?

File: handlers/customer/handler.go [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-	"database/sql"
+	"customerApi/stores"
 	"encoding/json"
 	"github.com/gorilla/mux"
 	"io"
 	"log"
 	"net/http"
 
-	"customerApi/drivers"
 	"customerApi/models"
```
Comment (aakanksha-jaiswal-zs) [inline handlers/customer/handler.go]: sort imports and do not use mixed caps for package, directory and file names

File: handlers/customer/handler.go [hunk 3]
```diff
@@ -33,9 +33,7 @@ func Create(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	_, err = db.Exec("INSERT INTO Customer (ID,NAME , PHONENO, ADDRESS) VALUES (?,?, ?, ?)",
-		c.ID, c.Name, c.PhoneNo, c.Address)
-
+	err = h.store.CreateCustomer(c)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
```
Comment (aakanksha-jaiswal-zs) [inline handlers/customer/handler.go]: Why BadRequest?

File: handlers/customer/handler.go [hunk 4]
```diff
@@ -53,39 +51,27 @@ func Create(w http.ResponseWriter, r *http.Request) {
...
 		w.WriteHeader(http.StatusNotFound)
-		_, err = w.Write([]byte("No Record Exists"))
+		_, err := w.Write([]byte("No Record Exists"))
 
 		if err != nil {
 			log.Println(err)
 		}
-	case nil:
+	case "No error":
 		resp, err := json.Marshal(c)
```
Comment (aakanksha-jaiswal-zs) [inline handlers/customer/handler.go]: Don't compare strings, compare error types

File: handlers/customer/handler.go [hunk 5]
```diff
@@ -94,14 +80,7 @@ func GetByID(w http.ResponseWriter, r *http.Request) {
...
-	db, err := drivers.ConnectToSQL()
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-	defer db.Close()
-
+func (h handler) UpdateByID(w http.ResponseWriter, r *http.Request) {
 	params := mux.Vars(r)
 	id := params["id"]
```
Comment (aakanksha-jaiswal-zs) [inline handlers/customer/handler.go:89]: covert ID to integer before passing it around

File: main.go [hunk 6]
```diff
@@ -1,25 +1,34 @@
...
 	"log"
 	"net/http"
 
 	"github.com/gorilla/mux"
 )
 
 func main() {
+	db, err := drivers.ConnectToSQL()
+	if err != nil {
+		log.Println("FATAL, Can't Connect to database")
```
Comment (aakanksha-jaiswal-zs) [inline main.go]: use `log.Fatalf()`

File: handlers/customer/handler_test.go [hunk 7]
```diff
@@ -1,132 +0,0 @@
-package customer
```
Comment (aakanksha-jaiswal-zs) [inline handlers/customer/handler_test.go:1]: handler tests are missing

File: stores/customer/store.go [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+		Scan(&c.ID, &c.Name, &c.PhoneNo, &c.Address)
+
+	switch err {
+	case sql.ErrNoRows:
+		return c, "No record"
+	case nil:
+		return c, "No error"
+	default:
+		return c, "Internal Server error"
+	}
```
Comment (aakanksha-jaiswal-zs) [inline stores/customer/store.go]: return custom error types, not strings

File: stores/customer/store.go [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+type store struct {
+	db *sql.DB
+}
+
+func New(db *sql.DB) stores.Customer {
+	return store{db: db}
+}
+
+func (s store) CreateCustomer(c models.Customer) error {
```
Comment (aakanksha-jaiswal-zs) [inline stores/customer/store.go]: `customer.CreateCustomer` looks redundant. `customer.Create` should be fine.

File: stores/interface.go [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package stores
+
+import "customerApi/models"
+
+type Customer interface {
+	CreateCustomer(c models.Customer) error
+	GetCustomer(id string) (models.Customer, string)
+	UpdateCustomer(id string, c models.Customer) error
+	DeleteCustomer(id string) error
```
Comment (aakanksha-jaiswal-zs) [inline stores/interface.go]: id should be integer , and Methods are in customer interface, so need not add customer suffix with each name

**Review by aakanksha-jaiswal-zs — CHANGES_REQUESTED** [review]

File: .golangci.yml (hunk 1 above)
Comment (ruchit-singh-zs) [reply .golangci.yml]: done

File: handlers/customer/handler.go (hunk 2 above)
Comment (ruchit-singh-zs) [reply handlers/customer/handler.go]: done

File: handlers/customer/handler.go (hunk 3 above)
Comment (ruchit-singh-zs) [reply handlers/customer/handler.go]: done

File: handlers/customer/handler.go (hunk 4 above)
Comment (ruchit-singh-zs) [reply handlers/customer/handler.go]: done

File: handlers/customer/handler.go (hunk 5 above)
Comment (ruchit-singh-zs) [reply handlers/customer/handler.go:89]: done

File: stores/customer/store.go (hunk 8 above)
Comment (ruchit-singh-zs) [reply stores/customer/store.go]: done

File: stores/customer/store.go (hunk 9 above)
Comment (ruchit-singh-zs) [reply stores/customer/store.go]: done

File: stores/interface.go (hunk 10 above)
Comment (ruchit-singh-zs) [reply stores/interface.go]: done

**Pull request closed without merging**